cdc/scheduler/internal/v3/agent/agent.go

```diff
@@ -185,22 +190,17 @@ func (a *agent) Tick(ctx context.Context, liveness model.Liveness) error {
 	return nil
 }
 
-const (
-	livenessSourceTick      = "tick"
-	livenessSourceHeartbeat = "heartbeat"
-)
-
-func (a *agent) handleLivenessUpdate(liveness model.Liveness, source string) {
-	if a.liveness != liveness {
-		if a.liveness == model.LivenessCaptureStopping {
+func (a *agent) handleLivenessUpdate(liveness model.Liveness) {
+	currentLiveness := a.liveness.Load()
+	if currentLiveness != liveness {
+		if currentLiveness == model.LivenessCaptureStopping {
 			// No way to go back, once it becomes shutting down,
 			return
 		}
-		log.Info("schedulerv3: agent liveness changed",
-			zap.String("old", a.liveness.String()),
-			zap.String("new", liveness.String()),
-			zap.String("source", source))
-		a.liveness = liveness
+		log.Info("schedulerv3: agent updates liveness",
+			zap.String("old", currentLiveness.String()),
+			zap.String("new", liveness.String()))
+		a.liveness.Store(liveness)
```

Comment:
Could there be a data race? For example, `a.liveness` is changed between the `Load()` and `Store()` calls by another thread?